### PR TITLE
Satisfying Simple Mob Combat

### PR DESCRIPTION
**Pull request by Geevies**
* Simple mobs now bleed. They also bleed out and die.
* Attacking a simple mob while unarmed will now use your default attack with a hitsound.
* Attacking a simple mob with an item will now properly show the attack animation and hitsound.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 1]
```diff
@@ -26,6 +26,11 @@
 	var/blood_state = BLOOD_NONE
 	var/image/blood_overlay
 
+	var/bleeding = FALSE
+	var/blood_amount = 50			// set a limit to the amount of blood it can bleed, otherwise it will keep bleeding forever and crunk the server
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/simple_animal.dm:30]: Don't think we need this. We can just make this a function of max_health and clamp it, no?

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
Why do we need to manually call handle_blood in so many cases instead of just doing it in Life()? I also have concerns about this making a lot of mobs easy to game, such as spiders with stairs.

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/mob/living/simple_animal/simple_animal.dm:30]: i don't get what you mean here

**Comment by Geevies** [comment]
> 
> 
> Why do we need to manually call handle_blood in so many cases instead of just doing it in Life()? I also have concerns about this making a lot of mobs easy to game, such as spiders with stairs.

the manual calls are just on update_icon because some mobs having floating animations, so the blood overlays need to be changed accordingly, the call itself is pretty rare. handle_blood IS done in Life, but that doesn't happen fast enough for icon changes to not look janky without the update_icon insert

as for making mobs easy to game, bleeding out a spider takes a long ass time, so if you want to spend forever doing it, be my guest

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by Geevies** [comment]
!review

**Review by Doxxmedearly — APPROVED** [review]

**Pull request merged**